src/cljc/athens/common_db.cljc

```diff
@@ -28,6 +28,28 @@
    :block/remote-id     {:db/unique :db.unique/identity}})
 
 
+(def v2-schema
+  {;; It would be nicer to have the reverse relationship here (:block/properties as a set
+   ;; of children refs, instead :block/property-of as a single parent ref), but that doesn't
+   ;; work with the tupleAttr below.
+   :block/property-of     {:db/cardinality :db.cardinality/one
+                           :db/valueType   :db.type/ref}
+   ;; key is to a property what order is to a child
+   :block/key             {:db/cardinality :db.cardinality/one
+                           :db/valueType   :db.type/ref}
+   ;; tupleAttrs are used here to ensure the relationship is unique,
+   ;; that there are no repeat keys in a block.
+   ;; https://github.com/tonsky/datascript/blob/master/docs/tuples.md
+   ;; It could later be extended to also ensure order is unique.
+   :block/property-of+key {:db/tupleAttrs [:block/property-of :block/key]
+                           :db/unique     :db.unique/identity}})
+
+
+;; TODO: generalize and use athens.self-hosted.migrate to migrate existing datascript dbs
+;; For now (the prototyping stage) it is enough that only new servers can use the new schema.
+(def schema (merge v1-schema v2-schema))
```

Comment:
Important: this PR cannot be merged before we have a migration set up for existing datascript dbs.